code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm

```diff
@@ -209,188 +158,189 @@
-					if(9)
-						//To make it more realistic, I added two gunshots (enough to kill)
-						src << 'sound/weapons/Gunshot.ogg'
-						spawn(rand(10,30))
+	if(client && feral >= 10) // largely a copy of handle_hallucinations() without the fake attackers. Unlike hallucinations, only fires once - if they're still feral they'll get hit again anyway.
+		spawn(rand(200,500)/(feral/10))
+			var/halpick = rand(1,100)
+			switch(halpick)
+				if(0 to 15) //15% chance
+					//Screwy HUD
+					//src << "Screwy HUD"
+					hal_screwyhud = pick(1,2,3,3,4,4)
+					spawn(rand(100,250))
+						hal_screwyhud = 0
+				if(16 to 25) //10% chance
+					//Strange items
+					//src << "Traitor Items"
+					if(!halitem)
+						halitem = new
+						var/list/slots_free = list(ui_lhand,ui_rhand)
+						if(l_hand) slots_free -= ui_lhand
+						if(r_hand) slots_free -= ui_rhand
+						if(istype(src,/mob/living/carbon/human))
+							var/mob/living/carbon/human/H = src
+							if(!H.belt) slots_free += ui_belt
+							if(!H.l_store) slots_free += ui_storage1
+							if(!H.r_store) slots_free += ui_storage2
+						if(slots_free.len)
+							halitem.screen_loc = pick(slots_free)
+							halitem.layer = 50
+							switch(rand(1,6))
+								if(1) //revolver
+									halitem.icon = 'icons/obj/gun.dmi'
+									halitem.icon_state = "revolver"
+									halitem.name = "Revolver"
+								if(2) //c4
+									halitem.icon = 'icons/obj/assemblies.dmi'
+									halitem.icon_state = "plastic-explosive0"
+									halitem.name = "Mysterious Package"
+									if(prob(25))
+										halitem.icon_state = "c4small_1"
+								if(3) //sword
+									halitem.icon = 'icons/obj/weapons.dmi'
+									halitem.icon_state = "sword1"
+									halitem.name = "Sword"
+								if(4) //stun baton
+									halitem.icon = 'icons/obj/weapons.dmi'
+									halitem.icon_state = "stunbaton"
+									halitem.name = "Stun Baton"
+								if(5) //emag
+									halitem.icon = 'icons/obj/card.dmi'
+									halitem.icon_state = "emag"
+									halitem.name = "Cryptographic Sequencer"
+								if(6) //flashbang
+									halitem.icon = 'icons/obj/grenade.dmi'
+									halitem.icon_state = "flashbang1"
+									halitem.name = "Flashbang"
+							if(client) client.screen += halitem
+							spawn(rand(100,250))
+								if(client)
+									client.screen -= halitem
+								halitem = null
+				if(26 to 35) //10% chance
+					//Flashes of danger
+					//src << "Danger Flash"
+					if(!halimage)
+						var/list/possible_points = list()
+						for(var/turf/simulated/floor/F in view(src,world.view))
+							possible_points += F
+						if(possible_points.len)
+							var/turf/simulated/floor/target = pick(possible_points)
+
+							switch(rand(1,3))
+								if(1)
+									//src << "Space"
+									halimage = image('icons/turf/space.dmi',target,"[rand(1,25)]",TURF_LAYER)
+								if(2)
+									//src << "Fire"
+									halimage = image('icons/effects/fire.dmi',target,"1",TURF_LAYER)
```

Comment:
The flick_overlay proc can replace most of this if you have suitable animations to go with. Fire would probably work. Not necessary.

Reply:
Hallucination code.